### ISSUE TITLE
doxywizard/config.cpp doesn't compile (Origin: bugzilla #154725)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#1288]
status RESOLVED severity *normal* in component *build* for *---*
Reported in version *1.3.x* on platform *Other*
Assigned to: Dimitri van Heesch

On 2004-10-06 20:34:14 +0000, Matthew Burgess wrote:

> in top-level source directory:
> 
> QTDIR=/opt/qt ./configure --prefix=/usr --with-doxywizard
> make
> 
> fails with:
> 
> make[3]: Entering directory `/sources/doxygen-1.3.9/addon/doxywizard'
> g++ -c -pipe -DENGLISH_ONLY -Wall -W -fno-exceptions -O2 -DNO_DEBUG
> -I../../src-I/opt/qt/include -o obj/doxywizard.o doxywizard.cpp
> g++ -c -pipe -DENGLISH_ONLY -Wall -W -fno-exceptions -O2 -DNO_DEBUG
> -I../../src-I/opt/qt/include -o obj/version.o version.cpp
> g++ -c -pipe -DENGLISH_ONLY -Wall -W -fno-exceptions -O2 -DNO_DEBUG
> -I../../src-I/opt/qt/include -o obj/inputstring.o inputstring.cpp
> g++ -c -pipe -DENGLISH_ONLY -Wall -W -fno-exceptions -O2 -DNO_DEBUG
> -I../../src-I/opt/qt/include -o obj/inputbool.o inputbool.cpp
> g++ -c -pipe -DENGLISH_ONLY -Wall -W -fno-exceptions -O2 -DNO_DEBUG
> -I../../src-I/opt/qt/include -o obj/inputstrlist.o inputstrlist.cpp
> g++ -c -pipe -DENGLISH_ONLY -Wall -W -fno-exceptions -O2 -DNO_DEBUG
> -I../../src-I/opt/qt/include -o obj/inputint.o inputint.cpp
> g++ -c -pipe -DENGLISH_ONLY -Wall -W -fno-exceptions -O2 -DNO_DEBUG
> -I../../src-I/opt/qt/include -o obj/expert.o expert.cpp
> flex -PconfigYY -t config.l >config.cpp
> g++ -c -pipe -DENGLISH_ONLY -Wall -W -fno-exceptions -O2 -DNO_DEBUG
> -I../../src-I/opt/qt/include -o obj/config.o config.cpp
> make: *** [doxygen] Error 2
> -su-3.00# <stdout>: In function `void configYY_init_buffer(yy_buffer_state*,
> FILE*)':
> <stdout>:2064: error: `isatty' undeclared (first use this function)
> <stdout>:2064: error: (Each undeclared identifier is reported only once for each
> function it appears in.)
> make[3]: *** [obj/config.o] Error 1
> make[3]: Leaving directory `/sources/doxygen-1.3.9/addon/doxywizard'
> make[2]: *** [all] Error 2
> make[2]: Leaving directory `/sources/doxygen-1.3.9/addon/doxywizard'
> make[1]: *** [all] Error 2
> make[1]: Leaving directory `/sources/doxygen-1.3.9'
> 
> It appears as though it's pulling in src/unistd.h instead of the system headers
> (-I../../src will be seen before the headers in /usr/include).  This means that
> anything that the declaration of isatty which is in /usr/include/unistd.h is
> never seen.
> 
> I can work around this by deleting src/unistd.h, or simply removing all it's
> contents.  That though will presumably break builds on Windows 95/NT.
> 
> Thanks,
> 
> Matt.

On 2004-10-07 19:36:02 +0000, Dimitri van Heesch wrote:

> Strange! My flex generated code does an external declaration of isatty() and as
> a result does not suffer from this problem. Which version of flex are you using?
> (flex -V). I use 2.5.4.

On 2004-10-07 19:45:09 +0000, Matthew Burgess wrote:

> I was hoping you weren't going to point to my flex installation - although it
> was my first suspect too.  It's 2.5.31 + a bunch of debian patches, available in
> a jumbo patch at
> http://www.linuxfromscratch.org/patches/downloads/flex/flex-2.5.31-debian-fixes-2.patch.
>  I can easily test with 2.5.4 though to see if that works for me.  Could you let
> me know what distro (and version) you use so I can grab the patches they apply,
> so we're testing with the exact same version.  Due to it's long-term lack of
> maintenance, flex must be one of the most heavily patched pieces of software I
> know of!
> 
> Thanks,
> 
> Matt.

On 2004-10-09 10:07:25 +0000, Dimitri van Heesch wrote:

> The unpatched 2.5.4 should work fine. 2.5.31 is known not to work (it generates
> wrong code!). I filed a bug report more than a year ago about this. I use SuSE
> 9.0 at the moment.

On 2006-01-31 23:55:28 +0000, Kevin McBride wrote:

> This bug is about the flex program, which GNOME does not provide the bugzilla for.  I am closing this bug as RESOLVED - NOTGNOME.

On 2006-02-03 00:13:08 +0000, Kevin McBride wrote:

> *** Bug 329320 has been marked as a duplicate of this bug. ***
